packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx

```diff
@@ -223,14 +223,107 @@ export const Attributes: FunctionComponent<AttributeProps> = props => {
 }
 
 export const BaseAttributes = (props: BaseProps) => {
-  const { feature, descriptions } = props
+  const { feature, descriptions, title = 'Attributes' } = props
   return (
-    <BaseCard {...props} title="Attributes">
+    <BaseCard {...props} title={title}>
       <Attributes {...props} attributes={feature} descriptions={descriptions} />
     </BaseCard>
   )
 }
 
+interface SubfeaturesToRenderProps {
+  title: string
+  attributes: Record<string, any>
+}
+
+/**
+ * Recursively parse features to extract all subfeatures.
+ * Return an subfeaturesToRender array of objects
+ */
+const getSubfeaturesToRender = (
+  features: Record<string, any>,
+  subfeaturesToRender: SubfeaturesToRenderProps[],
+): SubfeaturesToRenderProps[] => {
+  // Function who update subfeaturesToRender with the feature given, and recursively call
+  // getSubfeaturesToRender for subfeature inside feature.
+  const extractSubfeaturesToRender = (
+    feature: Record<string, any>,
+    subfeaturesToRenders: SubfeaturesToRenderProps[],
+  ) => {
+    let title = 'SubFeature'
+    title = feature.type
+    subfeaturesToRender.push({ title, attributes: feature })
+    // If subfeatures are present in feature, recursively call getSubfeaturesToRender
+    if ('subfeatures' in feature) {
+      getSubfeaturesToRender(feature.subfeatures, subfeaturesToRenders)
+    }
+    return subfeaturesToRender
+  }
+
+  if (Array.isArray(features)) {
+    for (const feature of features) {
+      extractSubfeaturesToRender(feature, subfeaturesToRender)
+    }
+  } else if (features !== undefined) {
+    extractSubfeaturesToRender(features, subfeaturesToRender)
+  }
+
+  return subfeaturesToRender
+}
+
+// Display a card named SUBFEATURES closed by default.
+// When open, only card with information about transcript and mRNA are displayed,
+// Accompanied with a button to load additionnal subfeature like exon/cds.
+export const BaseSubFeatures = (props: BaseProps) => {
+  const { feature, descriptions } = props
+  const [subfeaturesLoaded, setSubfeaturesLoaded] = useState(false)
+  let subfeaturesToRender = getSubfeaturesToRender(feature.subfeatures, [])
+  // If not loaded, display only transcript and mRNA
+  if (!subfeaturesLoaded) {
+    subfeaturesToRender = subfeaturesToRender.filter(subfeature =>
+      ['transcript', 'mRNA'].includes(subfeature.title),
+    )
```

Comment:
This filtering assumes a specific structure of subfeatures, which is not appropriate for this completely-generic feature detail implementation. Could you remove it?

Reply:
@rbuels If I understand correctly it should only display a `load subfeatures` button at start and then load every subfeatures ?
Example:
![image](https://user-images.githubusercontent.com/24433720/100219305-19965280-2f16-11eb-8427-ebda2bea1f29.png)

![image](https://user-images.githubusercontent.com/24433720/100219176-f9669380-2f15-11eb-895a-8d55768e280a.png)

Reply:
What I meant is, this is specific to "transcript" and "mRNA" subfeatures. This feature track is used to display many things, not just genes. Filtering here for only mRNA and transcript subfeatures means that other types of features won't be able to display their subfeatures. I see you removed it in 9588d98 anyway, so you probably understand what I mean already. :-)